File: go/2.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	c := 0
	ret := &ListNode{-1, nil}
	pos := ret
	for l1 != nil || l2 != nil || c != 0 {
		v := c
		if l1 != nil {
			v = v + l1.Val
			l1 = l1.Next
		}
		if l2 != nil {
			v = v + l2.Val
			l2 = l2.Next
		}
		c = v / 10
		v = v % 10
		pos.Next = &ListNode{v, nil}
		pos = pos.Next
	}
	return ret.Next
}

File: py/2.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        c = 0
        ret = ListNode(-1)
        pos = ret
        while l1 or l2 or c :
            v = c
            if l1 :
                v += l1.val
                l1 = l1.next
            if l2 :
                v += l2.val
                l2 = l2.next
            c = v / 10
            v %= 10
            pos.next = ListNode(v)
            pos = pos.next
        return ret.next
            
